File: launcher/gen_images.php
<?php
$output = '
typedef struct {
    size_t size;
    const uint8_t data[];
} binfile_t;

';

chdir(__DIR__.'/images');

foreach (glob('*.png') as $file) {
    $data = file_get_contents($file);
    $size = strlen($data);
    $hex = [];
    for ($x = 0; $x < $size; ++$x) {
        $hex[] = sprintf('0x%02X', ord($data[$x]));
    }
    $hexdata = wordwrap(implode(', ', $hex), 100);
    $struct_name = preg_replace('/[^a-z0-9]+/i', '_', basename($file, '.png'));

    $output .= "static const binfile_t $struct_name = {.size = $size, .data = {\n$hexdata\n} };";
    $output .= "\n\n";
}

file_put_contents('../main/images.h', $output);

File: launcher/gen_images.py
import os, sys, glob, textwrap

# This scripts should be run to update images.h when you edit images

os.chdir(sys.path[0])

output  = """
typedef struct {
    size_t size;
    const uint8_t data[];
} binfile_t;

"""

for file in glob.glob("images/*.png"):
    with open(file, "rb") as f:
        data = f.read()
        size = len(data)
        struct_name = os.path.basename(file)[0:-4]
        hexdata = ""
        for c in data:
            hexdata += "0x%02X, " % c

        hexdata = "\n".join(textwrap.wrap(hexdata[0:-2], 100))

        output += "static const binfile_t %s = {" % struct_name
        output += ".size = %d, .data = {\n%s\n} " % (size, hexdata)
        output += "};\n\n"

with open("main/images.h", "w", newline='') as f:
    f.write(output)
